### PR TITLE
Fix issue where email addresses are case sensitive

### PR DESCRIPTION
**Pull request by Staartvin**
1) Registering will now lowercase the email address.
2) Signing in with an email address is now also case insensitive.

### CODE REVIEW

File: module/User/src/User/Mapper/User.php [hunk 1]
```diff
@@ -60,11 +60,11 @@ public function findByLogin($login)
         if (is_numeric($login)) {
             $qb->where('u.lidnr = ?1');
         } else {
-            $qb->where('m.email = ?1');
+            $qb->where('m.email LIKE ?1');
         }
 
         // set the parameters
-        $qb->setParameter(1, $login);
+        $qb->setParameter(1, $login = str_replace("%", "", $login));
```
Comment (jszanto) [inline module/User/src/User/Mapper/User.php]: I think the `$login = ` doesn't quite belong here. Also, single quotes > double quotes.

File: module/User/src/User/Mapper/User.php [hunk 2]
```diff
@@ -60,11 +60,11 @@ public function findByLogin($login)
         if (is_numeric($login)) {
             $qb->where('u.lidnr = ?1');
         } else {
-            $qb->where('m.email = ?1');
+            $qb->where('m.email LIKE ?1');
```
[the remark this answers is not in the document]
Comment (jszanto) [reply module/User/src/User/Mapper/User.php]: Good suggestion!

**Comment by armandduijn** [comment]
@jszanto Do we need to lowercase all the existing emails in the database?

**Comment by Koen1999** [comment]
> 
> 
> @jszanto Do we need to lowercase all the existing emails in the database?

I don't think that's neccesary with this code.

I think that we can actually merge this sooner or later

I do have one question for @Staartvin though
What does this piece of code do?
``str_replace('%', '', $login)``

File: module/User/src/User/Mapper/User.php [hunk 3]
```diff
@@ -60,11 +60,11 @@ public function findByLogin($login)
         if (is_numeric($login)) {
             $qb->where('u.lidnr = ?1');
         } else {
-            $qb->where('m.email = ?1');
+            $qb->where('m.email WHERE LOWER(m.email) = ?1');
         }
 
         // set the parameters
-        $qb->setParameter(1, $login);
+        $qb->setParameter(1, str_replace('%', '', $login));
```
Comment (jszanto) [inline module/User/src/User/Mapper/User.php]:
```suggestion
        $qb->setParameter(1, strtolower($login));
```

File: module/User/src/User/Model/NewUser.php [hunk 4]
```diff
@@ -119,7 +119,7 @@ public function setLidnr($lidnr)
      */
     public function setEmail($email)
     {
-        $this->email = $email;
+        $this->email = strtolower($email);
```
Comment (jszanto) [inline module/User/src/User/Model/NewUser.php]: This doesn't seem needed if we just do it on the fly.

**Comment by Koen1999** [comment]
When is case sensitivity of email addresses actually a problem?

**Comment by Koen1999** [comment]
Please refer to #915 from here. It's the continuation of the PR, so we can solve the issue

**Pull request closed without merging**